server/v2/types.go

```diff
@@ -0,0 +1,19 @@
+package serverv2
+
+import (
+	"github.com/spf13/viper"
+
+	coreapp "cosmossdk.io/core/app"
+	"cosmossdk.io/core/transaction"
+	"cosmossdk.io/log"
+	"cosmossdk.io/server/v2/appmanager"
+)
+
+type AppCreator[T transaction.Tx] func(log.Logger, *viper.Viper) AppI[T]
+
+type AppI[T transaction.Tx] interface {
+	GetAppManager() *appmanager.AppManager[T]
```

Comment:
can we create and expected interface here for appmanager? best not to tie it to our implementation

Reply:
let's do it in a follow up, other deps is removed now though